kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/model/VirtualCluster.java

```diff
@@ -187,11 +187,13 @@ private Optional<SslContext> buildDownstreamSslContext() {
     }
 
     private Optional<SslContext> buildUpstreamSslContext() {
-        return targetCluster.tls().map(tls -> {
+        return targetCluster.tls().map(targetClusterTls -> {
             try {
-                var sslContextBuilder = Optional.ofNullable(tls.key()).map(NettyKeyProvider::new).map(NettyKeyProvider::forClient).orElse(SslContextBuilder.forClient());
-                var withTrust = Optional.ofNullable(tls.trust()).map(NettyTrustProvider::new).map(tp -> tp.apply(sslContextBuilder))
+                var sslContextBuilder = Optional.ofNullable(targetClusterTls.key()).map(NettyKeyProvider::new).map(NettyKeyProvider::forClient)
+                        .orElse(SslContextBuilder.forClient());
+                var withTrust = Optional.ofNullable(targetClusterTls.trust()).map(NettyTrustProvider::new).map(tp -> tp.apply(sslContextBuilder))
                         .orElse(sslContextBuilder);
+                withTrust.endpointIdentificationAlgorithm("HTTPS");
```

Comment:
This change brings forward the Netty 4.2 change. If we don't want to support Hostname validation (arguably a breaking change) we can pass `null` here instead. 

I'd argue we should enable it, but if there are concerns about backwards compatibility introduce an experimental config property to either specify the algorithm or just a boolean to disable it. Basically an opt out that we carry for a couple of releases to allow us to deprecate in on our usual time line

Reply:
+1 for an opt out boolean, maybe even a top level config rather than per-VC